// **************************************************
// Protocol Buffers specification of this API service
// **************************************************

// Syntax version (version of Protocol Buffers language)
syntax = "proto3";

// **** Service definition ****
// 1 method/ endpoint
// takes a message of the "GetUserRequest" type and returns another message of the "User" type
service ExampleService {
    rpc GetUser (GetUserRequest) returns (User);
    rpc GetStatus (GetStatusRequest) returns (Status);
}

message Status {
    string status = 1;
    string validity = 2;
}

message GetStatusRequest {
    string status = 1;
}

// **** Message type 1 ****
// 3 string type fields
// number after each field is the field number --> for uniquely identifying the field
message User {
    string name = 1;
    string display_name = 2;
    string email = 3;
}

// **** Message type 2 ****
message GetUserRequest {
    string name = 1;
}